src/main/java/uk/gov/hmcts/darts/common/repository/UserAccountRepository.java

```diff
@@ -51,4 +51,14 @@ AND securityRole.id IN (:roleIds)
         """)
     List<UserAccountEntity> findByEmailAddressOrAccountGuidForRolesAndGlobalAccessIsTrue(String emailAddress, String accountGuid, Set<Integer> roleIds);
 
+    @Query("""
+        SELECT DISTINCT userAccount
+        FROM UserAccountEntity userAccount
+        JOIN userAccount.securityGroupEntities securityGroup
+        JOIN securityGroup.securityRoleEntity securityRole
+        WHERE userAccount.id = :userId
+        AND securityRole.id = :securityRole
+        """)
+    UserAccountEntity findByRoleAndUserId(Integer securityRole, Integer userId);
```

Comment:
I thought best practice was to make the return type Optional, but yes I can see `findSystemUser` doesn't follow that either... 

[Best Practices (Backend)](https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1694706043)
* 3.2 Repository Methods Should Return Optional<> For All Non-Collection Return Types

Reply:
changed to Optional